api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/swagger/ApiDocRetrievalServiceRest.java

```diff
@@ -266,123 +67,37 @@ private String getApiDocUrl(ApiInfo apiInfo, InstanceInfo instanceInfo, RoutedSe
      * @return the information about ApiDoc content as application/json
      * @throws ApiDocNotFoundException if the response is error
      */
-    private String getApiDocContentByUrl(@NonNull String serviceId, String apiDocUrl) throws IOException {
+    private Mono<String> getApiDocContentByUrl(@NonNull String serviceId, String apiDocUrl) {
         HttpGet httpGet = new HttpGet(apiDocUrl);
         httpGet.setHeader(org.apache.http.HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
 
-        return secureHttpClientWithoutKeystore.execute(httpGet, response -> {
-                String responseBody = "";
-                var responseEntity = response.getEntity();
-                if (responseEntity != null) {
-                    responseBody = EntityUtils.toString(responseEntity, StandardCharsets.UTF_8);
-                }
-
-            if (HttpStatus.SC_OK == response.getCode()) {
-                return responseBody;
-            } else {
-                    throw new ApiDocNotFoundException("No API Documentation was retrieved due to " + serviceId +
-                        " server error: '" + responseBody + "'.");
-                }
+        // TODO: refactor with reactive client
+        return Mono.defer(() -> {
+            try {
+                return Mono.just(secureHttpClientWithoutKeystore.execute(httpGet, response -> {
```

Comment:
better to use `Mono.fromCallable` rather than `Mono.just` as the execute method could throw some exception

Reply:
I guess since it is wrapped by Mono.defer it is no necessary.